### ISSUE TITLE
feat: HuggingFace TokenClassification + ONNX export for browser NER

### ISSUE THREAD

**Issue by HikaruEgashira** [HikaruEgashira/pleno-anonymize#25]
## 概要

pleno-audit Chrome拡張のDLP機能でブラウザ内PII検出を実現するため、HuggingFace Transformers + ONNX + Transformers.js パイプラインを構築する。

## 背景

現行のspaCy CNNモデルはThinc独自形式のため、ブラウザでの推論再現が困難（HashEmbedの4-hash sum + with_array内部状態管理）。標準エコシステム（Transformers.js）に移行することで、この問題を根本解決する。

### 1. データ変換（完了済み）

`convert_to_hf_dataset.py` で既存JSONデータをHuggingFace Dataset（BIOタグ）に変換。

```bash
make convert-hf-v02
```

- 入力: `data/raw/ja-v02/generated.json`
- 出力: `data/hf/ja-v02/` (train=6826, dev=853, test=854)
- トークナイザ: `ku-nlp/deberta-v2-base-japanese`（SentencePiece、ブラウザ互換）

### 2. Fine-tune（RunPod GPU）

`train_hf.py` で `ku-nlp/deberta-v2-base-japanese` をTokenClassificationでfine-tune。

```bash
make train-hf-v02
```

- ベースモデル: `ku-nlp/deberta-v2-base-japanese`
- タスク: TokenClassification (BIO, 11クラス)
- ラベル: ADDRESS, BANK_ACCOUNT, DATE_OF_BIRTH, ORGANIZATION, PERSON
- 推奨: RunPod A4000/A5000 GPU、5エポック、batch_size=16

### 3. ONNX Export + INT8量子化

`export_onnx.py` で学習済みモデルをONNX + INT8ダイナミック量子化。

```bash
make export-onnx-v02
```

- 出力: `output/hf-ja-v02-onnx/model_quantized.onnx`
- 推定サイズ: ~110MB (INT8)

### 4. HuggingFace Hub Push

`0xhikae/ja-ner-onnx` にpush。Transformers.jsから直接利用可能に。

## ベースモデル選定理由

| モデル | トークナイザ | ブラウザ互換 | 日本語品質 |
|--------|-------------|-------------|-----------|
| tohoku-nlp/bert-base-japanese-v3 | MeCab + WordPiece | **不可**（MeCab必須） | 高 |
| **ku-nlp/deberta-v2-base-japanese** | **SentencePiece** | **可**（tokenizer.json あり） | **高** |
| xlm-roberta-base | SentencePiece | 可 | 低（多言語） |

## pleno-audit側の統合（別リポ）

```typescript
import { pipeline } from '@huggingface/transformers';
const ner = await pipeline('token-classification', '0xhikae/ja-ner-onnx', { dtype: 'q8' });
const entities = await ner('田中太郎は東京都千代田区に住んでいます');
```

現行のカスタム推論エンジン（dlp-local-ner.ts, dlp-tokenizer.ts, WASM tokenizer）を全廃止。

## 依存追加

`pyproject.toml` に `[project.optional-dependencies] hf` を追加済み:
- transformers, datasets, seqeval, accelerate, optimum[onnxruntime]

```bash
uv sync --extra hf
```

## Makefile

```bash
make all-hf-ja-v02  # convert → train → export 全工程
```

**Comment by HikaruEgashira** [comment]
DeBERTa v2 base Japaneseでの再トレーニング完了。

## 変更点
- ベースモデルを `tohoku-nlp/bert-base-japanese-v3` (MeCab) → `ku-nlp/deberta-v2-base-japanese` (SentencePiece) に変更
- `export_onnx.py` に `--push-to` オプション追加 (HuggingFace Hub push)
- `huggingface-hub` 依存追加

## トレーニング結果
- **Test F1: 0.9719** (Precision: 0.9703, Recall: 0.9735)
- 5エポック、batch_size=16、FP16、RTX A4000

## アーティファクト
- ONNX量子化モデル (163.5MB): https://huggingface.co/0xhikae/ja-ner-onnx
- SentencePieceトークナイザー同梱 → Transformers.jsから直接利用可能

**Comment by HikaruEgashira** [comment]
## ベンチマーク結果: base vs tiny

| モデル | パラメータ | Test F1 | Precision | Recall | ONNX INT8 |
|--------|-----------|---------|-----------|--------|-----------|
| `deberta-v2-base` | 110M | **0.9719** | 0.9703 | 0.9735 | 163.5MB |
| `deberta-v2-tiny` | 13.9M | 0.9216 | 0.9058 | 0.9380 | **8.9MB** |

### 結論
baseモデル (163.5MB) はブラウザには大きすぎるため、**tinyモデル (8.9MB, F1=0.92) に切り替え**。
F1の差は0.05だが、Chrome拡張のDLP用途では十分な精度。

### 現在のアーティファクト
- https://huggingface.co/0xhikae/ja-ner-onnx (tiny, 8.9MB INT8)
- トレーニング設定: 20エポック, batch_size=16, lr=3e-5, warmup=0.15

### pleno-audit統合
```typescript
import { pipeline } from '@huggingface/transformers';
const ner = await pipeline('token-classification', '0xhikae/ja-ner-onnx', { dtype: 'q8' });
```
